go/lib.h

```diff
@@ -46,10 +50,18 @@ typedef void (*FailureCallback)(uintptr_t channel_address,
  *
  * The returned `ConnectionResponse` should be manually freed by calling `free_connection_response`, otherwise a memory leak will occur. It should be freed whether or not an error occurs.
  *
+ * `connection_request_bytes` is an array of bytes that will be parsed into a Protobuf `ConnectionRequest` object.
```

Comment:
you've decided to keep the protobuf dependency? why?

Reply:
Constructing the address list in Go was particularly difficult. @aaron-congo can probably elaborate on that. Maybe we just need to pass the appropriate information to Rust and have it construct the address list instead.

Reply:
please elaborate - the protobuf dependency affects users, and avoiding it is one of the upsides of not using a socket.